### PR TITLE
InRepoConfigGitCache: cleans up after merge conflict

### PR DESCRIPTION
**Pull request by chaodaiG**

### CODE REVIEW

**Comment by fejta** [comment]
the `git add -A` is important, otherwise you won't delete new files.

**Comment by chaodaiG** [comment]
> the `git add -A` is important, otherwise you won't delete new files.
> 
> Aborting non-normal modes also seems important. Did you test out these scenarios? AKA a merge conflict.

yeah I had tested locally, that `git reset --hard HEAD` cleans up after merge conflict

File: prow/git/git.go [hunk 1]
```diff
@@ -263,6 +263,31 @@ func (r *Repo) Clean() error {
 	return os.RemoveAll(r.dir)
 }
 
+// ResetHard runs `git reset --hard`
+func (r *Repo) ResetHard(commitlike string) error {
+	// `git reset --hard` doesn't cleanup untracked file, adding them to be safe
```
Comment (petr-muller) [inline prow/git/git.go:268]: You could also `git clean -fd` for more straightforward cleanup (just a suggestion)

Comment (chaodaiG) [reply prow/git/git.go]: feels like better to understand than `add -A`, updated the PR to use `clean -df`

**Review by petr-muller — APPROVED** [review]

**Pull request merged**